src/pestuary.py

```diff
@@ -46,22 +50,58 @@
 except ApiException as e:
     print("Exception when calling UserApi->user_api_keys_get: %s\n" % e)
 
+@click.group()
+def cli():
+    pass
+
+@cli.result_callback()
+def process_result(result):
+    print(json.dumps(result, default=lambda x: x.to_dict()))
+    return result
+
+
+
+#wrote magic to automate some things that clck doesnt do, i didnt want to have to specify each argument to click for each method when it can be done automatically using decorators
+def magic(fn):
+    if __name__ != "__main__":
+        return fn
+    argspec = inspect.getargspec(fn)
+    arguments = argspec.args
+    defaults = argspec.defaults
+    if not arguments:
+        arguments = []
+    if not defaults:
+        defaults = []
+    requireds = [index < len(arguments) - len(defaults)  for index in range(len(arguments)) ]
+    for i in range( len(arguments) - 1, -1, -1) :
+        fn = click.argument(arguments[i], required=requireds[i])(fn)
+    fn = cli.command()(fn)
+    return fn
```

Comment:
This is very clever but not readable. We better delete this decorator and decorate the functions with the default click decorators manually. For instance, here's how `collection_create` would look:

```python
@click.command()
@click.option('--name', required=True, help="The collection's name")
@click.option('--description', default='', help="The collection's description string")
def collection_create(name, description=''):
    body = swagger_client.MainCreateCollectionBody(name=name, description=description)
    return collectionsApi.collections_post(body)
```

Reply:
It doesn't make sense to have the function decorator on every line of code in pestuary.py. it will make managing that much much more complicated, especially as the AP changes down the line since this codebase is meant to integrate with the swagger auto generated api. Having this little bit of code in pestuary_cli that accomplishes the goal of having a command line tool that's executable from bash, and doesn't need to be maintained. Overall the readability of the project is much better by having a complicated but powerful function in a separate module that doesn't interact with anything, than having many lines of code for each api function that will change as the api changes. Also i'm working on pushing my code upstream to the click project if they'll accept it, then I can refactor that into an import. Other wise I can set ths up as it's own separately maintained project for turning modules into CLIs